server/jms/src/main/java/org/hawkular/apm/server/jms/CommunicationDetailsDeriverMDB.java

```diff
@@ -49,7 +48,7 @@
     private TracePublisherJMS tracePublisher;
 
     @Inject
-    private CommunicationDetailsPublisher communicationDetailsPublisher;
+    private CommunicationDetailsPublisherJMS communicationDetailsPublisher;
```

Comment:
I am probably just not understanding this. If you inject this, you should have a producer somewhere that can give you the concrete type of `CommunicationDetailsPublisher`. Why do you then specify the concrete type here? Remember that JMS and MDB are api/abstractions over the concrete message oriented middleware or the data source creating the message. There are resource adapters other than JMS that can drive MDBs